src/components/top-menu/top-menu.ts

```diff
@@ -0,0 +1,195 @@
+    });
+
+    const found = new BaseComponent('div', { className: 'top_menu__left__item found' });
+    found.setContent('Found: 0');
+    topMenuLeft.append(found);
+
+    const topMenuRight = new BaseComponent('div', { className: 'top_menu__right' });
+    this.container.append(topMenuRight);
+
+    this.changeColumnsView = new BaseComponent('div', {
+      className: 'top_menu__right__item columns_view',
+    });
+    topMenuRight.append(this.changeColumnsView);
+
+    this.changeView = new BaseComponent('div', { className: 'top_menu__right__item view' });
+    topMenuRight.append(this.changeView);
+
+    this.favouritesLink = new Link({
+      ...favouritesLink,
+      className: 'top_menu__right__item favourites_link',
+    });
+    topMenuRight.append(this.favouritesLink);
+
+    const favourites = new BaseComponent('div', { className: 'top_menu__right__item' });
+    favourites.setContent('Favourites');
+    this.favouritesLink.append(favourites);
+
+    const sort = new BaseComponent('div', { className: 'top_menu__right__item sort' });
+    sort.setContent('Sort');
+    topMenuRight.append(sort);
+
+    this.dropDownSort = new BaseComponent('span', { className: 'drop_down' });
+    sort.append(this.dropDownSort);
+
+    const dropDownMenuSort = new BaseComponent('ul', { className: 'drop_down_menu sort' });
+    this.dropDownSort.append(dropDownMenuSort);
+
+    const menuItemPriceDown = new BaseComponent('li', { className: 'menu_item price_down' });
+    menuItemPriceDown.setContent('by Price Down');
+    dropDownMenuSort.append(menuItemPriceDown);
+    const menuItemPriceUp = new BaseComponent('li', { className: 'menu_item price_up' });
+    menuItemPriceUp.setContent('by Price Up');
+    dropDownMenuSort.append(menuItemPriceUp);
+
+    this.dropDownSort.getNode().addEventListener('click', () => {
+      dropDownMenuSort.toggleClass('active');
+      menuItemPriceDown.toggleClass('active');
+      menuItemPriceUp.toggleClass('active');
+    });
+
+    const search = new BaseComponent('div', { className: 'top_menu__right__item search' });
+    search.setContent('Search');
+    topMenuRight.append(search);
+
+    const magnifyingGlass = new BaseComponent('span', { className: 'magnifying_glass' });
+    search.append(magnifyingGlass);
+
+    this.changeViewButton();
+
+    this.changeColumnsViewButton();
+
+    Link.addNavigationLink(this.favouritesLink);
+    Link.addNavigationLink(this.allLink);
+  }
+
+  changeViewButton() {
+    for (let i = 0; i < 8; i++) {
+      const changeViewElement = new BaseComponent('span', { className: 'element' });
+      this.changeView.append(changeViewElement);
+      this.changeView.getNode().addEventListener('click', () => {
+        changeViewElement.toggleClass('active');
+      });
+    }
+    this.changeView.getNode().addEventListener('click', () => {
+      this.onclick?.();
+    });
+  }
+
+  changeColumnsViewButton() {
+    for (let i = 0; i < 12; i++) {
```

Comment:
Don't use magic numbers, create constant with descriptive naming for 8, because it is impossible to understand, why `12`, not `128`.